sdk/trace/trace_test.go

```diff
@@ -217,13 +217,26 @@ func TestSetName(t *testing.T) {
 	}
 }
 
-func TestRecordingIsOn(t *testing.T) {
-	tp := NewTracerProvider()
-	_, span := tp.Tracer("Recording on").Start(context.Background(), "StartSpan")
-	defer span.End()
-	if span.IsRecording() == false {
-		t.Error("new span is not recording events")
-	}
+func TestSpanIsRecording(t *testing.T) {
+	t.Run("is recording on", func(t *testing.T) {
```

Comment:
This could be a good place to use table-driven tests.  Look at the tests on either side of this one, they range over a set of test cases that have a name, some configuration, and expected results.  If the logic for each test is the same, which it is here, parameterizing test cases like this makes it easier to test new scenarios.